File: SDE_sheet/Arrays/22_largest_subarray_with_kSum/solution.java
class GfG{
    int maxLen(int arr[], int n){
        int sum = 0, max = 0;
        Map<Integer, Integer> map = new HashMap<>();
        for(int i=0;i<n;i++){
            sum += arr[i];
            if(sum == 0) max = i+1;
            if(map.containsKey(sum)){
                max = Math.max(max, i - map.get(sum));
                
            }
            if(!map.containsKey(sum)) map.put(sum, i);
        }
        return max;
    }
}

File: SDE_sheet/Arrays/22_largest_subarray_with_kSum/solution.cs
public class Solution{
    public int maxLen(int[] arr, int k){
        int sum = 0, max = 0;
        Dictionary<int, int> map = new Dictionary<int, int>();
        map[0] = -1;
        
        for (int i = 0; i < k; i++){
            sum += arr[i];
            if (map.ContainsKey(sum)){
                max = Math.Max(max, i - map[sum]);
            }
            if (!map.ContainsKey(sum)){
                map[sum] = i;
            }
        }
        
        return max;
    }
}
